newrelic_python_agent/plugins/mysql.py

```diff
@@ -890,12 +914,16 @@ def poll(self):
         self.initialize()
         self.raw_metrics = dict()
         try:
-            self.connection = self.connect()
-            cursor = self.connection.cursor()
-            self.collect_stats(cursor)
-            cursor.close()
-            self.connection.close()
+            # open a new connection
+            with self.connect() as cursor:
+                # self.verify_uuid(cursor)
+                # self.logger.debug("done verifying uuid")
+                self.collect_stats(cursor)
+                self.logger.debug("done collecting data")
+            # build stats
             self.add_stats()
+        except ValueError as err:
+            self.logger.error(err)
```

Comment:
I think using `logging.exception` instead of `logging.error` might be better here since it would incorporate the exception information